ui/src/ElasticSearch.elm

```diff
@@ -0,0 +1,114 @@
+module ElasticSearch exposing (..)
+
+import Http
+import Json.Decode as Decode
+import Json.Encode as Encode
+import Types exposing (..)
+
+
+filterToJsonProperty : String -> Maybe String
+filterToJsonProperty filter =
+    case filter of
+        "product" ->
+            Just "source.product"
+
+        "channel" ->
+            Just "target.channel"
+
+        "locale" ->
+            Just "target.locale"
+
+        "version" ->
+            Just "target.version"
+
+        "platform" ->
```

Comment:
Hmm, this function has a code smell to me. My first thought is that filters should be an enum. But then I looked at the context where this is used and I wonder if maybe this is even necessary? Why don't we just use the JSON names at the call site?